### PR TITLE
Add an integration test for consistency with common-definitions

### PR DESCRIPTION
**Pull request by danielhuppmann**
Remove GDP variables following the merging of https://github.com/IAMconsortium/common-definitions/pull/54 and add an integration-test to ensure consistency between common-definitions & legacy definitions.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
@phackstock, one minor issue in the test is that `nomenclature.yaml` is not deleted if the test fails. I tried briefly playing around with some try-except-finally but it become a bit complicated... And there isn't really any harm from that file not being deleted when running the test locally.

File: tests/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        },
+        "definitions": {
+            "region": {"repository": "common-definitions"},
+            "variable": {"repository": "common-definitions"},
+        },
+    }
+
+    with open(config_file, "w") as file:
+        yaml.dump(config, file)
+
```
Comment (phackstock) [inline tests/test_integration.py:24]:
```suggestion
    try:
```

File: tests/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    nomenclature.DataStructureDefinition("definitions")
```
Comment (phackstock) [inline tests/test_integration.py]:
```suggestion
        nomenclature.DataStructureDefinition("definitions")
```

File: tests/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (phackstock) [inline tests/test_integration.py:27]:
```suggestion
    finally:
```

File: tests/test_integration.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    pathlib.Path(config_file).unlink()
```
Comment (phackstock) [inline tests/test_integration.py]:
```suggestion
        pathlib.Path(config_file).unlink()
```

**Review by phackstock — CHANGES_REQUESTED** [review]
Looks good to me.
Regarding the test not deleting the config file, using a `try... finally` in the test should solve the issue. I've put a solution that should work as suggestions.
Then you could also remove the entry from the .gitignore file.

**Comment by danielhuppmann** [comment]
Seems to work as expected, thanks for the hint!

File: .gitignore [hunk 5]
```diff
@@ -1,5 +1,7 @@
 ### common-definitions ###
 common-definitions
+nomenclature.yaml
```
Comment (phackstock) [inline .gitignore]:
```suggestion
```

**Review by phackstock — APPROVED** [review]
Now the entry in the .gitignore is no longer needed. 
Other than that good to merge :+1:

**Pull request merged**